### DISCUSSION TITLE
Wikidata and scientific events

### DISCUSSION THREAD

**Discussion by Simon-Cobb** [SmartDataAnalytics/OpenResearch#127] [Ideas]
New project for scientific events in Wikidata: https://rq.bitplan.com/index.php/WikiData_Scientific_Events

**Comment by WolfgangFahl** [comment]
Hi Simon - welcome!
It's awesome to have you here. I love what you did so far. 
Wolfgang

**Reply by Simon-Cobb** [reply]
What is the next step? Can we agree on some data to upload?

**Reply by Simon-Cobb** [reply]
Can you provide or agree on a full dataset that I can work on? It is more efficient for me to work on a large dataset. My availability changes and there is a risk that data ingest will not be completed if my time is taken by other priorities.

I will post a suggested dataset on https://rq.bitplan.com/index.php/WikiData_Scientific_Events for review. Look forward to your comments and discussing with you next week.

**Reply by Simon-Cobb** [reply]
Is there a mapping between Open Research Event Series and conferences in DBLP (i.e. https://dblp.org/db/conf/xxxxx)? I'm matching on acronym and validating with text comparison of event title using Python difflib and ngram fingerprints. Have reviewed before marking as matched but I have limited knowledge of these conference series. Shall I add columns with the results of each comparison?

**Reply by WolfgangFahl** [reply]
Unfortunately the mapping between OpenResearcch and dblp event series is still very limited. https://www.openresearch.org/wiki/List_of_DblpEventSeries shows only 40 out of some 1000 event series to have such a mapping. 

Trying to map by acronym will lead to the disambiguation of events issue #42 which we so far try to solve with approaches like getting a full "signature" by using the Proceedings Title Parser #113. We didn't try out the full potential of Natural Language Processing yet as #122 Weaviate would offer us.

This week i could not to work on this issue due to other priorities but i hope before our meeting next week the situation will change and I can give you better input. Is there an option for trying out your current approach without changing the official wikidata entries yet?

**Reply by Simon-Cobb** [reply]
Thanks, I thought that was the case with OpenResearch.

Still working on the data - I've identified 258 event series with matching ngram and score > 0.9 from libdiff that are not in Wikidata. Currently validating Microsoft Academic matches and will share when completed.

**Comment by WolfgangFahl** [comment]
I think it is better if we make the links we are using public:
```
wikipush -l -s  rq -t or -p WikiData_Scientific_Events 
copying 1 pages from rq to or
1/1 ( 100%): copying WikiData_Scientific_Events ...✅
wikipush -l -s  rq -t or -p Template:OR
copying 1 pages from rq to or
1/1 ( 100%): copying Template:OR ...✅
wikipush -l -s  rq -t or -p "PTP Minimum viable dataset"
copying 1 pages from rq to or
1/1 ( 100%): copying PTP Minimum viable dataset ...✅
wikipush -l -s rq -t or -p File:Dblp-conf.xlsx
```
So 
https://www.openresearch.org/wiki/WikiData_Scientific_Events
would be the page to start from.

The excel sheet  looks interesting. I am especially happy that your are considering the OpenResearch links. Potentially it would be better if you'd go from on of the 8 OPENRESEARCH copies that we set up because we can clean up data there much more efficiently and effectively than in the legacy production wiki. 

What cleanups would you need or benefit from?

I am especially thinking about the list from https://www.openresearch.org/wiki/PTP_Minimum_viable_dataset

* title
* acronym
* series
* homepage (+ archive link and state (blue link/red link)
* ordinal
* year
* month
* country
* region
* city
* startdate
* enddate
* frequency
* event type

**Reply by Simon-Cobb** [reply]
There are 186 series without a title in the data I have from OpenResearch - adding titles would be the obvious task to help with reconciliation. The other thing I noticed are a few questionable series, e.g. https://www.openresearch.org/wiki/ICPAD_2009, which might be an event in a series or single event. But neither are urgent.

I have another 169 matches between DBLP and OpenResearch that appear to be correct but do not score as high in text comparison. However, I want to do further checks to validate before adding to the spreadsheet.

Can upload a batch later this evening but happy to wait until next weekend as working on other tasks. Likely to be very busy again this week so apologies in advance if I am not responsive.

**Reply by Simon-Cobb** [reply]
Also, I have over 2,000 matches between DBLP and Microsoft Academic - a lot of validation work is required as the quality of Microsoft Academic data varies.

**Reply by Simon-Cobb** [reply]
If the dblp-conf.xlsx file is public, it should include a attribution to the Microsoft Academic source data: https://zenodo.org/record/2628216. However, it does not and I am also slightly concerned that it lacks descriptive metadata. Can it be restricted - perhaps only accessible to users with an account or from a specific link? It's more of a working document that final dataset.

**Reply by WolfgangFahl** [reply]
I'have removed the file from the public access as requested. The matching problem is very interesting and I am keen to discuss and document the details of it together with you. Thank you for all the work so far!

**Reply by Simon-Cobb** [reply]
Thanks. I've uploaded a new version and description including sources of the data. It can be made public if you like but the data will be on Wikidata anyway - that's is my preferred way to publish it.

Do you have OpenRefine? If so, I can give you the project to extract the expressions, although not all of the facets will have been captured.

**Reply by Simon-Cobb** [reply]
Output of small test ingest: https://w.wiki/3274

**Reply by WolfgangFahl** [reply]
On the OPENRESEARCH side off things we need a way to automatically add the wikidata backlinks which we are working on. There is also a need for a rating on how sure we are the linked data is correctly linked. During our meeting we might want to discuss these indicators.

**Comment by WolfgangFahl** [comment]
Simon at https://or.bitplan.com/index.php/List_of_EventSeries there is a copy of all the data of december 2020 where the title problem seems not to be visible.

**Reply by WolfgangFahl** [reply]
I have now added #136 and will tell the curators about it

**Comment by WolfgangFahl** [comment]
Next step:

- DBLP series and event import to wikidata can be done independently of OPENRESEARCH - so if you like please go ahead with it
- for OPENRESEARCH the current sample of the pre-migration state is great and good
- doing the OPENRESEARCH wikidata match should be postponed until the [migration](https://github.com/SmartDataAnalytics/OpenResearch/labels/Migration) is done

We 'd love to try https://www.linguee.com/german-english/translation/rasterfahndung.html - techniques using the https://www.openresearch.org/wiki/PTP_Minimum_viable_dataset signature elements.

**Reply by Simon-Cobb** [reply]
Why are you using SMW rather than a Wikibase?

**Reply by WolfgangFahl** [reply]
I'd even love to use a proper graph database. In the end Wikibase is based on blazegraph and that would allow you to have a SPARQL endpoint and Gremlin Access and what not at the same time. Unfortunately the blazegraph people were aqui-hired by Amazon and now work on Neptune which is not OPEN source and not an option for our Research project at this time. Wikibase has it own oddities and quirks and IMHO is only a vehicle on a way to true graph database based solutions. Quite a few people are already using Neo4j, Virtuoso and other databases and Wikibased technically also can do that. I love the proposal of Laura Ham #122 since that would give us more options together with the https://pypi.org/project/pylodstorage/ approach for standard and legacy access with e.g. csv, json and rdf.

In short the technical detail issues hold us up in getting the benefits we'd love to get ...

**Reply by Simon-Cobb** [reply]
I've added a few more conference series to Wikidata - sign into the app to edit directly in tabular format: https://tabernacle.toolforge.org/?#/tab/sparql/SELECT%20%3Fseries%0A%7B%0A%20%20%3Fseries%20wdt%3AP31%20wd%3AQ47258130%0A%7D/Len%3BDen%3BP31%3BP1476%3BP1813%3BP664%3BP8926%3BP5127%3BP6366%3BP973

Haven't forgotten about sharing the OpenRefine project and will do so when I finish uploading. Hopefully later today but, if not, tomorrow.

**Reply by WolfgangFahl** [reply]
It's the first time i see tabernacle at work which shows that i still have to learn a lot in terms of available tools. I like the addition of the dblp provenance data. I might be able to do some more checking this afternoon. I look forward to get a copy of the openrefine project that you have created for the import. It is a very valuable asset especially in many ways. Personally i love the prospect of being able to use it as a kind of "tutorial" for how to use openrefine in our usecase.

**Comment by WolfgangFahl** [comment]
For the event import one of the aspect we are quite interested in is the link to the next upcoming event. I have seen that Microsoft academic supplies this e.g. the next upcoming [CVPR](https://academic.microsoft.com/conference/1158167855/publication/search?q=CVPR%3A%20Computer%20Vision%20and%20Pattern%20Recognition&qe=And(Composite(C.CId%253D1158167855)%252CTy%253D%270%27)&f=&orderBy=0) is available as an .ics Calendar entry 
![grafik](https://user-images.githubusercontent.com/1336221/109410767-99144180-799d-11eb-867c-4ecc0ad52678.png)

The idea would be that we'll add the CVPR 2021 entry to:
https://www.openresearch.org/wiki/CVPR 

As soon as the data on an upcoming event is available. Is there a tool that would trigger us if the next upcoming event is entered in wikidata? At this point i think entering future events in wikidata is a rarely done. Have you seen some examples of future events?

**Reply by Simon-Cobb** [reply]
Yes, I've seen future events in Wikidata. As you say, it's not common but that is to be expected as the coverage of events in general is not so good. An example is WikidataCon 2021: https://www.wikidata.org/wiki/Q73768833

I like the idea of adding upcoming events. Could use significant event (P793) for such information as call for papers (Q286706) and paper submission deadline  (Q47483633). Or proposing new properties would be an option if these events are to be added regularly.

**Reply by Simon-Cobb** [reply]
Have started matching event series to proceeding series and events. Also matching proceeding series to proceedings. Will post queries when I have made some progress with the addition of these relationships.

**Reply by Simon-Cobb** [reply]
Have added DNB, VIAF, LCNAF etc. identifiers to the conference series from DBLP: [Conference series in Wikidata](https://tabernacle.toolforge.org/?#/tab/sparql/SELECT%20%3Fseries%0A%7B%0A%20%20%3Fseries%20wdt%3AP31%20wd%3AQ47258130%0A%7D/Len%3BDen%3BP31%3BP1476%3BP1813%3BP527%3BP664%3BP8926%3BP214%3BP227%3BP244%3BP5127%3BP6366%3BP973).

Also started the ingestation of conference events, e.g. https://reasonator.toolforge.org/?q=Q105700510, and working on identifying existing items for proceedings and proceedings series.

**Comment by csisc** [comment]
To proceed, you can align DBLP publications with Wikidata items using external identifiers. One of the ways to do this action is to use Wikidata hub to return the Wikidata item corresponding to DOIs. An example of how this works is https://hub.toolforge.org/P356:10.1016/J.JBI.2019.103292?format=json. As you see here, the result is returned in the JSON Format and can consequently be used in Python bots. You can find attached to this email message a Python code converting a list of DOIs in the plain text format to a list of corresponding Wikidata identifiers. You can adapt this method to include other external identifiers such as ISBN and ISSN.

**Reply by WolfgangFahl** [reply]
@csisc  - thank you for the hint. I'll look into this.

**Comment by WolfgangFahl** [comment]
@Simon-Cobb  - thank you for the OpenRefine Video you let me record in one of the zoom sessions we had. Today we had a project meeting where six member of the team watched the video and discussed how we are going to use OpenRefine for the cleanup of the OPENRESEARCH data. Your input is much appreciated and we decided to try out OpenRefine for the parts of our migration effort where this seems to make sense based on your input. We appreciate your participation in our efforts a lot!

**Comment by nichtich** [comment]
To better illustrate the usage of mapping to Wikidata you should add links to Scholia, e.g.

* https://scholia.toolforge.org/event-series/Q105700510 (conference series)
* https://scholia.toolforge.org/event/Q105891131 (conference)

I recommend to use [Scholia Favicon](https://github.com/fnielsen/scholia/tree/master/scholia/app/static/favicon) to link to Scholia.

**Comment by WolfgangFahl** [comment]
Simon for #162 I'd love to have the conference entries for ECIR YYYY and link them to the proceedings. I'd love to start with ECIR 2013

**Reply by Simon-Cobb** [reply]
No problem, I'll add the ECIR conferences as you suggest. Won't have much time during the week but can work on it at the weekend

**Reply by WolfgangFahl** [reply]
@Simon-Cobb  excellent  - i started with https://www.wikidata.org/wiki/Q40874627 (Advances in Information Retrieval: 35th European Conference on IR Research, ECIR 2013, Moscow, Russia, March 24-27, 2013. Proceedings) to show how https://www.wikidata.org/wiki/Property:P4745 (is proceedings from) may be used see https://diagrams.bitplan.com/render/svg/0x589f1ec0.svg. So https://www.wikidata.org/wiki/Q106460755 (ECIR 2013) was created by me manual as an example how publications/proceedings of an event an the actual event might be linked.

**Comment by WolfgangFahl** [comment]
@Simon-Cobb today i presented your work in our project meetings see https://rq.bitplan.com/index.php/ConfIDent_Migrationplan/Overview#OpenRefine - i'd love to present part of this also at EMWCon this week. Is that generally o.k. with you? Would you like to participate in presenting?

**Reply by WolfgangFahl** [reply]
https://rq.bitplan.com/index.php?title=Wikipush_Toolkit_Application_in_the_ConfIDent_Project&action=slide
